### PR TITLE
Upgrade to HB.beta.1

### PR DESCRIPTION
**Pull request by tib**
- add scripts
- upgrade to HB.beta.1
- add github workflow

### CODE REVIEW

File: .github/workflows/run-checks.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 1
+    
+    - name: Install Dependencies
+      run: |
+        brew install mint
+        mint install NickLockwood/SwiftFormat@0.48.17 --no-link
```
Comment (adam-fowler) [inline .github/workflows/run-checks.yml:23]: HB is using v0.51.15 now, v0.48.17 is quite old

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+jobs:
+  
+  tests:
+    runs-on: self-hosted
+    # runs-on: ubuntu-latest
+    # strategy:
+    #   matrix:
+    #     image:
+    #       - 'swift:5.9'
```
Comment (adam-fowler) [inline .github/workflows/run-tests.yml:21]: 5.10?

**Review by adam-fowler — APPROVED** [review]
HB changes all look good

File: .github/workflows/run-checks.yml (hunk 1 above)
Comment (tib) [reply .github/workflows/run-checks.yml]: Thanks. Now using the latest release.

File: .github/workflows/run-tests.yml (hunk 2 above)
Comment (tib) [reply .github/workflows/run-tests.yml]: Added both 5.9 and 5.10 to the test matrix + running self-hosted macOS tests.

**Pull request merged**